client/src/components/CodeEditor.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import AceEditor from 'react-ace';
+
+import "ace-builds/src-noconflict/mode-python";
+import "ace-builds/src-noconflict/mode-java";
+import "ace-builds/src-noconflict/mode-c_cpp";
+import "ace-builds/src-noconflict/mode-javascript";
+import "ace-builds/src-noconflict/mode-ruby";
+import "ace-builds/src-noconflict/theme-twilight";
+import "ace-builds/src-noconflict/theme-tomorrow";
+
+const CodeEditor = ({ language, theme, value, onChange }) => {
+  const defaultLanguage = 'python';
+  const defaultValue = '';
+
+  let aceTheme;
+  switch (theme) {
+    case 'dark':
+      aceTheme = 'twilight';
+      break;
+    case 'light':
+    default:
+      aceTheme = 'tomorrow';
+      break;
+  }
+
+  return (
+    <AceEditor
+      fontSize={16}
+      mode={language || defaultLanguage}
+      theme={aceTheme}
+      value={value || defaultValue}
+      hightlightActiveLine={true}
+      showPrintMargin={false}
+      width='100%'
+      onChange={onChange}
+      setOptions={{ useWorker: false }}
+      style={{ borderRadius: '5px' }}
```

Comment:
Could you expand a bit on this? I'm not sure what you mean.

Reply:
Oh I mean instead of doing inline styling, to use this - https://material-ui.com/styles/basics/#hook-api - unless AceEditor does not take **className**

Reply:
Ah, I see. I wasn't having success wrapping the component in a div and applying border-radius that way but I neglected to try applying css to the component directly. Fixed to work as suggested now.